File: common/src/main/java/mconi/common/sim/OniServices.java
package mconi.common.sim;

import java.util.concurrent.atomic.AtomicBoolean;

/**
 * Central service registry for ONI simulation systems.
 */
public final class OniServices
{
	private static final AtomicBoolean INITIALIZED = new AtomicBoolean(false);
	private static final OniSimulationRuntime SIMULATION_RUNTIME = new OniSimulationRuntime();

	private OniServices() { }

	public static void bootstrap()
	{
		if (INITIALIZED.compareAndSet(false, true))
		{
			SIMULATION_RUNTIME.bootstrap();
		}
	}

	public static OniSimulationRuntime simulationRuntime()
	{
		return SIMULATION_RUNTIME;
	}
}

File: common/src/main/kotlin/mconi/common/sim/OniServices.kt
package mconi.common.sim

import java.util.concurrent.atomic.AtomicBoolean

/**
 * Central service registry for ONI simulation systems.
 */
object OniServices {
    private val initialized = AtomicBoolean(false)
    private val simulationRuntime = OniSimulationRuntime()

    @JvmStatic
    fun bootstrap() {
        if (initialized.compareAndSet(false, true)) {
            simulationRuntime.bootstrap()
        }
    }

    @JvmStatic
    fun simulationRuntime(): OniSimulationRuntime = simulationRuntime
}
